pkg/runtime/bulk_subscriber.go

```diff
@@ -259,80 +266,98 @@ func (a *DaprRuntime) createEnvelopeAndInvokeSubscriber(ctx context.Context, psm
 	if err != nil {
 		id = idObj.String()
 	}
-	psm.cloudEvents = psm.cloudEvents[:psm.length]
-	psm.rawData = psm.rawData[:psm.length]
-	psm.entries = psm.entries[:psm.length]
+	psm.pubSubMessages = psm.pubSubMessages[:psm.length]
+	psm.path = path
 	envelope := runtimePubsub.NewBulkSubscribeEnvelope(&runtimePubsub.BulkSubscribeEnvelope{
 		ID:       id,
 		Topic:    topic,
-		Entries:  psm.rawData,
 		Pubsub:   psName,
 		Metadata: msg.Metadata,
 	})
+	var firstAttempt atomic.Bool
+	firstAttempt.Store(true)
+	_, err = policy(func(ctx context.Context) (any, error) {
+		if !firstAttempt.Load() {
+			newPubSubMsg := utils.Filter(psm.pubSubMessages, func(ps pubSubMessage) bool {
+				index := (*entryIdIndexMap)[ps.entry.EntryId]
+				return (*bulkResponses)[index].Error != nil
+			}, firstAttempt.Load())
```

Comment:
```suggestion
			}, false)
```

Or, rather we will remove this since the `Filter` method is changing.

Reply:
Yup, removed.